pmm-app/src/pmm-settings/Settings.service.ts

```diff
@@ -1,10 +1,53 @@
-import { apiRequest } from '../shared/components/helpers/api';
+import { apiRequest } from 'shared/components/helpers/api';
+import { showSuccessNotification } from 'shared/components/helpers';
+import { Messages } from './Settings.messages';
+import { Settings } from './Settings.types';
+
+export type LoadingCallback = (value: boolean) => void;
+export type SettingsCallback = (settings: Settings) => void;
 
 export const SettingsService = {
-  getSettings() {
-    return apiRequest.post<any, any>('/v1/Settings/Get', {});
+  async getSettings(setLoading: LoadingCallback, setSettings: SettingsCallback) {
+    let response: any;
+
+    try {
+      setLoading(true);
+      response = await apiRequest.post<any, any>('/v1/Settings/Get', {});
+      setSettings(toModel(response.settings));
+    } catch (e) {
+      console.error(e);
+    } finally {
+      setLoading(false);
```

Comment:
I would use `setSettings` and `setLoading` outside this function and use `getSettings` just to fetch data